testutil/environment/environment.go

```diff
@@ -162,6 +162,11 @@ func (e *Execution) IsUserNamespace() bool {
 	return root != ""
 }
 
+// RuntimeIsWindowsContainerD returns whether containerd runtime is used on Windows
+func (e *Execution) RuntimeIsWindowsContainerD() bool {
+	return os.Getenv("DOCKER_WINDOWS_CONTAINERD_RUNTIME") == "1"
```

Comment:
`DOCKER_WINDOWS_CONTAINERD_RUNTIME` is only used now when running the tests from the CI script (or if manually set). Is it possible for this (and the two uses in `integration-cli/requirements_test.go`) to use the same logic as `system.InitContainerdRuntime`? Can this access the _Docker Under Test_ command-line?

Reply:
Does `docker info` (or the `/info` endpoint) return information about what runtime is used? Perhaps we should move the "containerd" part from `daemon.fillPlatformInfo` into a separate function, so that we provide the same information on Windows (if we're running with containerd as runtime); https://github.com/moby/moby/blob/7b9275c0da707b030e62c96b679a976f31f929d3/daemon/info_unix.go#L59-L68


Same for `daemon.fillPlatformVersion`; https://github.com/moby/moby/blob/7b9275c0da707b030e62c96b679a976f31f929d3/daemon/info_unix.go#L175-L184

Reply:
No `docker info` does not contain runtime info currently but I think that it should be part of #42089 as it updates that logic and after that is done CI can also ask it from daemon that which runtime it is using instead of that environment variable.